src/os/bluestore/BlueStore.cc

```diff
@@ -14201,6 +14205,44 @@ void BlueStore::_pad_zeros(
   ceph_assert(bl->length() == length);
 }
 
+void BlueStore::_detect_zeros_and_write_small(
+    uint64_t loffs,
+    BlobRef b,
+    uint64_t blob_len,
+    uint64_t o,
+    bufferlist& bl,
+    uint64_t o0,
+    uint64_t len0,
+    bool _marked_unused,
+    bool _new_blob,
+    WriteContext *wctx,
+    uint64_t chunk_size,
+    bool reuse_blob, // whether or not to log a "reuse blob" message
+    bool write_small_unused) // whether or not to inc l_bluestore_write_small_unused
+{
+    // If the bufferlist contains only zeroes, we logically extend
+    // the space without writing the zeroed bufferlist to the disk for better optimization.
+    if (!bl.is_zero()) {
+      // There is relevant data in this bufferlist, so write to disk.
+      _pad_zeros(&bl, &o, chunk_size);
+
+      if (reuse_blob) {
```

Comment:
as you're changing this piece of code I would suggest to uniform this  logging with _do_write_big and log some notice about writing unconditionally. And similarly indicate relevant reuse_blob value.

Reply:
I addressed this comment, but see if you agree with the change in the most recent commit.
I realized that the "reuse_blob" parameter I had added actually communicates the same information as the existing "_new_blob" parameter (if _new_blob is true, a new blob is used; if _new_blob is false, an existing blob is reused). I updated the log message to print unconditionally, but using the "_new_blob" parameter instead of "reuse_blob".